src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -3802,5 +4537,53 @@ mod tests {
             tug.update();
             tug
         }
+
+        fn test_supplied_power(bus_id: ElectricalBusType, is_powered: bool) -> SuppliedPower {
+            let mut supplied_power = SuppliedPower::new();
+            match bus_id {
+                ElectricalBusType::AlternatingCurrent(1)
+                | ElectricalBusType::AlternatingCurrent(2)
+                | ElectricalBusType::AlternatingCurrentEssential
+                | ElectricalBusType::AlternatingCurrentEssentialShed
+                | ElectricalBusType::AlternatingCurrentGndFltService
+                | ElectricalBusType::AlternatingCurrentStaticInverter => {
+                    supplied_power.add(
+                        bus_id,
+                        if is_powered {
+                            Potential::single(
+                                PotentialOrigin::EngineGenerator(1),
+                                ElectricPotential::new::<volt>(115.),
+                            )
+                        } else {
+                            Potential::none()
+                        },
+                    );
+                }
+
+                ElectricalBusType::DirectCurrent(1)
+                | ElectricalBusType::DirectCurrent(2)
+                | ElectricalBusType::DirectCurrentBattery
+                | ElectricalBusType::DirectCurrentEssential
+                | ElectricalBusType::DirectCurrentEssentialShed
+                | ElectricalBusType::DirectCurrentGndFltService
+                | ElectricalBusType::DirectCurrentHot(1)
+                | ElectricalBusType::DirectCurrentHot(2) => {
+                    supplied_power.add(
+                        bus_id,
+                        if is_powered {
+                            Potential::single(
+                                PotentialOrigin::Battery(1),
+                                ElectricPotential::new::<volt>(28.),
+                            )
+                        } else {
+                            Potential::none()
+                        },
+                    );
+                }
+                _ => panic!("Bus not registered"),
+            }
+
+            supplied_power
+        }
```

Comment:
Does the potential origin or voltage really matter for testing? If not, then you can reduce this whole block of code to:

```rust
supplied_power.add(
    bus_id,
    if is_powered {
        Potential::single(
            PotentialOrigin::EngineGenerator(1),
            ElectricPotential::new::<volt>(115.),
         )
     } else {
         Potential::none()
     },
);
```

Reply:
Nope indeed, was afraid to be slapped by engineers doing this, or to trigger elec issues with stupid values. If it's not the case simpler is better!